config/hooks.go

```diff
@@ -0,0 +1,445 @@
+	*method = HttpMethod(i)
+	return nil
+}
+
+// UnmarshalJSON ValidationType
+func (typ *ValidationType) UnmarshalJSON(arg []byte) error {
+	i, err := validationTypeEnum.UnMarshal(arg)
+	if err != nil {
+		return errors.Wrap(err, "Failed to unmarshal validationTypeEnum")
+	}
+	*typ = ValidationType(i)
+	return nil
+}
+
+// UnmarshalJSON FailLevel
+func (lvl *FailLevel) UnmarshalJSON(arg []byte) error {
+	i, err := failLevelEnum.UnMarshal(arg)
+	if err != nil {
+		return errors.Wrap(err, "Failed to unmarshal failLevelEnum")
+	}
+	*lvl = FailLevel(i)
+	return nil
+}
+
+// Validate hook settings, returns list of warnings or error
+func (hook *Hook) Validate() ([]string, error) {
+	if hook.Url == "" {
+		return nil, errors.Errorf("hook defined with empty URL")
+	}
+
+	// Validate extractors
+	for _, extractor := range hook.Extractors {
+		if err := extractor.Validate(); err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	return nil, nil
+}
+
+// Validate extractor
+func (extractor *Extractor) Validate() error {
+	if extractor == nil {
+		return nil
+	}
+	if extractor.Name == "" {
+		return errors.New("no key name defined for extractor")
+	}
+	if extractor.Path.String() == "" {
+		return errors.New("no path defined for extractor")
+	}
+
+	return extractor.Validator.Validate()
+}
+
+//Validate validator settings
+func (validator *Validator) Validate() error {
+	if validator == nil {
+		return nil
+	}
+
+	switch validator.Type {
+	case ValidationTypeBool:
+		if _, isBool := validator.Value.(bool); !isBool {
+			return errors.Errorf("validator value<%v> not of type bool", validator.Value)
+		}
+	case ValidationTypeNumber:
+		if _, isFloat := validator.Value.(float64); !isFloat {
+			return errors.Errorf("validator value<%v> not of type number", validator.Value)
+		}
+	case ValidationTypeString:
+		if _, isString := validator.Value.(string); !isString {
+			return errors.Errorf("validator value<%v> not of type string", validator.Value)
+		}
+	case ValidationTypeNone:
+	default:
+		return errors.Errorf("ValidationType<%v> not supported", validator.Type)
+	}
+	return nil
+}
+
+// Execute hook
+func (hook *Hook) Execute(ctx context.Context, logEntry *logger.LogEntry, data *hookData, allowUntrusted bool) error {
+	hook.init()
+
+	payload, err := hook.Content.ExecuteString(data)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	headers, err := hook.Headers.Execute(data)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	buf := helpers.GlobalBufferPool.Get()
+	defer helpers.GlobalBufferPool.Put(buf)
+	_, err = buf.WriteString(payload)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, strings.ToUpper(httpMethodEnum.StringDefault(int(hook.Method), http.MethodPost)), hook.Url, buf)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	req.Header.Set("Content-Type", hook.ContentType)
+	for k, v := range headers {
+		req.Header.Set(k, v)
+	}
+
+	// TODO try avoid creating client here
```

Comment:
It will only be done once pre and once post test though, right? So should be a small overhead

Reply:
Correct. But if we need to change any of the settings at some point we might forget to change this one too.